source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -221,39 +214,56 @@ bool FaultFilter::faultOverflow() {
   return false;
 }
 
-bool FaultFilter::isDelayEnabled() {
+bool FaultFilter::isDelayEnabled(const Http::RequestHeaderMap& request_headers) {
   const auto request_delay = fault_settings_->requestDelay();
   if (request_delay == nullptr) {
     return false;
   }
 
+  const auto percentage_header =
```

Comment:
From a perf perspective it would be optimal to not do this lookup unless we are actually using a header provider (similar elsewhere), especially because right now this is going to be an O(N) scan to find the header. Can you figure out a way to only do the lookup on demand? Perhaps by passing the entire map and initializing the header provider with the header to look for?

Reply:
Good idea, I updated all headers providers to work like this - we pass headers map and we do a specific header lookup only if a header provider is used. After my recent changes, there should be no performance impact related to header lookup at all if fault filter is not configured to use header faults.

One more question regarding the implementation of the percentage header providers. As it is now, we try to read a percentage header (such as `x-envoy-fault-abort-request-percentage`) and if it's not present or its value is invalid we fallback to the value that's specified in proto config file. Do you think that falling back to `100%` would make more sense (as opposed to looking at proto config file)?

https://github.com/envoyproxy/envoy/blob/ed6fa7f15454ee74ab5e9ba21ae55942a5da895d/source/extensions/filters/common/fault/fault_config.cc#L14-L37

Reply:
> Do you think that falling back to 100% would make more sense (as opposed to looking at proto config file)?

I think falling back to the config is more flexible, as the operator can configure a default of 100%, 0%, etc. if they want?

Reply:
That was my thought too 👍  Leaving it as it is then.